File: pod/14sep.py

def main():
    fnum = int(input())
    height = int(input())
    dia = int(input())
    k = int(input())
    l = []
    for i in range(k):
        x = int(input())
        y = int(input())
        if(x == dia):
            l.append(y)
    cost = 0;
    l.sort()
    while(len(l) > 1):
        x = l[0]
        l.pop(0)
        y = l[0]
        l.pop(0)
        cost += (x+y);
        l.append(x+y)
        l.sort()
    print(cost)
    return 0;

main()

File: college_life/pod/14sep.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
    int fnum;
    int height;
    int dia;
    int k;
    cin>>fnum>>height>>dia>>k;
    int x,y;
    priority_queue<int,vector<int>,greater<int>> pq;
    for(int i = 0; i < k; i++){
        cin>>x>>y;
        if(x == dia){
            pq.push(y);
        }
    }
    int cost = 0;
    while(pq.size() > 1){
        int x = pq.top();
        pq.pop();
        int y = pq.top();
        pq.pop();
        cost += (x+y);
        pq.push(x+y);
    }
    cout<<cost<<endl;
    return 0;
}